### PR TITLE
NEMO-5289 "Category" Column on Product Listing shows 'undefined' when product linked categories are deleted

### PR DESCRIPTION
**Pull request by IrvinFan**

### CODE REVIEW

File: core/app/models/spree/taxon.rb [hunk 1]
```diff
@@ -86,5 +88,9 @@ def child_index=(idx)
     def touch_parent
       parent.touch if parent
     end
+
+    def touch_products
+      products.map(&:touch)
```
Comment (mtuckergd) [inline core/app/models/spree/taxon.rb]: Can we please wrap this in a `delay_touch`?

**Comment by marcheiligers** [comment]
And add a test :smile:

**Comment by IrvinFan** [comment]
Made suggested changes

**Comment by mtuckergd** [comment]
:+1:

**Pull request merged**